src/engine/renderer/tr_main.cpp

```diff
@@ -2353,40 +2353,57 @@ void R_AddLightInteractions()
 	bspNode_t    *leaf;
 	link_t       *l;
 
+	dynamicLightRenderer_t dynamicLightRenderer = dynamicLightRenderer_t( r_dynamicLightRenderer.Get() );
+
 	tr.refdef.numShaderLights = 0;
+
 	for ( i = 0; i < tr.refdef.numLights; i++ )
 	{
 		light = tr.currentLight = &tr.refdef.lights[ i ];
 
-		if ( light->isStatic ) {
-			if ( r_staticLight->integer != 1 || ( ( r_precomputedLighting->integer || tr.worldLight != lightMode_t::MAP ) && !light->noRadiosity ) )
+		if ( light->isStatic )
+		{
+			if ( !glConfig2.staticLight )
 			{
-				if( r_staticLight->integer == 2 ) {
```

Comment:
The old code checks for 1 and 2 as values of r_staticLight but it was changed to bool.

Reply:
This is on purpose the very first reason of this PR.

When the tiled renderer was made default in d3b1f7fbd6a3cc11dac15718323ff88340140534, it was needed to modify two cvars, `r_staticLight` and `r_dynamicLight`. Each cvar were both selecting a level of feature _AND_ a renderer. When I modified the default value of `r_dynamicLight` I failed to see it was also needed to set `r_staticLight`. Enabling or disabling the tiled renderer required to modify two cvars, but the same cvars were also used to enable or disable the features.

Now the `r_staticLight` and `r_dynamicLight` are only booleans enabling or not those features, and `r_dynamicLightRenderer` is selecting wich renderer we want for both.

Reply:
For reference this code is the code to render the static lights with the dynamic renderer, which was and is still mostly a copy-paste of the code rendering dynamic lights with the dynamic renderer (see my other comment on that topic).

If the tiled renderer cannot run because of a missing feature, both those static lights rendered dynamically and the dynamic lights can't be rendered with that renderer.